src/views/ArticleView.vue

```diff
@@ -0,0 +1,354 @@
+			section.remove();
+		}
+	},
+	'remove "See also" "Notes" section': ( doc ) => {
+		const seeAlsoH2 = doc.querySelector( '#See_also' );
+		if ( seeAlsoH2 ) {
+			seeAlsoH2.parentElement.parentElement.remove();
+		}
+		const NotesH2 = doc.querySelector( '#Notes' );
+		if ( NotesH2 ) {
+			NotesH2.parentElement.parentElement.remove();
+		}
+	},
+	'turn figure into an image': ( doc ) => {
+		// enable lazyload image
+		const imageSpanNodes = doc.querySelectorAll( 'span.pcs-lazy-load-placeholder' );
+		for ( const imageSpanNode of imageSpanNodes ) {
+			const source = imageSpanNode.getAttribute( 'data-src' );
+			const image = document.createElement( 'img' );
+			image.src = source;
+			image.height = imageSpanNode.getAttribute( 'data-height' );
+			imageSpanNode.parentNode.appendChild( image );
+
+			if ( !imageSpanNode.parentNode.classList.contains( 'image' ) ) {
+				imageSpanNode.parentNode.classList += ' image';
+			}
+
+			imageSpanNode.remove();
+		}
+	},
+	'turn sections into collapsible section': ( doc ) => {
+		const sections = doc.querySelectorAll( 'section' );
+		for ( let i = 1; i < sections.length; i++ ) {
+			sections[ i ].style.removeProperty( 'display' );
+		}
+
+		const sectionTitle = doc.querySelectorAll( 'h2' );
+		for ( let i = 0; i < sectionTitle.length; i++ ) {
+			const section = sectionTitle[ i ].parentElement.parentElement;
+			section.style.display = 'unset';
+			section.classList.add( 'collapsible' );
+		}
+	}
+};
+
+const addEvents = function ( doc ) {
+	// add sections collapsible toggle events
+	for ( const section of doc.querySelectorAll( '.collapsible' ) ) {
+		section.firstChild.onclick = function () {
+			section.classList.toggle( 'collapsible' );
+		};
+	}
+};
+
+const fetchArticle = function ( title ) {
+	const url = `https://en.wikipedia.org/api/rest_v1/page/mobile-html/${title}`;
+
+	return fetch( url ).then( ( resp ) => {
+		if ( resp.ok ) {
+			return resp.text();
+		} else {
+			throw new Error();
+		}
+	} ).then( ( html ) => {
+		// transform response html object
+		const doc = new DOMParser().parseFromString( html, 'text/html' );
+		Object.keys( transforms ).forEach( ( key ) => {
+			transforms[ key ]( doc );
+		} );
+
+		return doc.body.outerHTML;
+	} ).then( ( html ) => {
+		contentHtml.value = html;
+	} ).then( () => {
+		addEvents( contentRef.value );
+	} );
+
+};
+
+const fetchRelatedArticle = function ( title ) {
+	const articleObject = getArticle( title );
+
+	if ( articleObject ) {
+		const category = articleObject.category;
+		const allRelatedArticles = categories[ category ].filter( ( c ) => c.title !== title );
+		return allRelatedArticles.slice( 0, 3 );
+
+	}
+};
+
+const goTo = function ( title ) {
+	fetchArticle( title ).then( () => {
+		window.scrollTo( 0, 0 );
+		relatedArticles.value = fetchRelatedArticle( title );
+	} );
+};
+
+onMounted( function () {
+	goTo( route.params.title );
+} );
+
+</script>
+
+<style>
```

Comment:
Could you add "scoped" to this style so it doesn't override everything

Reply:
that's true, but scoped doesn't work for global style like <body>, let me see what i can do here.

Reply:
good catch, done.